### PR TITLE
[Remove Stream/Optional usage] LPS-176099 Remove_Stream_usages_for_module_sharing_api

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina, Dante,

This is for ticket https://issues.liferay.com/browse/LPS-176099.
Tried to use TransformUtil for filtering and transforming the array.
Thanks for checking.

### CODE REVIEW

**Comment by zhangyan0701** [comment]
ci:test:sharing

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/sharing/sharing-api/src/main/java/com/liferay/sharing/security/permission/SharingEntryAction.java [hunk 1]
```diff
@@ -39,14 +39,17 @@ public enum SharingEntryAction {
...
 
-		return Stream.of(
-			values()
-		).filter(
-			sharingEntryAction ->
-				(sharingEntryAction.getBitwiseValue() & bitwiseValue) != 0
-		).collect(
-			Collectors.toList()
-		);
+		return TransformUtil.transformToList(
```
Comment (dantewang) [inline modules/apps/sharing/sharing-api/src/main/java/com/liferay/sharing/security/permission/SharingEntryAction.java:42]: @jeffwu0724 To make the code simpler, use `ListUtil.filter(Arrays.asList(values()), sharingEntryAction -> ...)`.

Comment (dantewang) [reply modules/apps/sharing/sharing-api/src/main/java/com/liferay/sharing/security/permission/SharingEntryAction.java:42]: @jeffwu0724 `Arrays.asList()` creates an `Arrays`-private implementation of array based list, which directly wraps the passed array, so no copy is involved in this step. Thus, using this, you achieve the same, with less lines of code.

**Pull request closed without merging**